### PR TITLE
[7023] fix for catalog_service_roles, no longer a set (main)

### PR DESCRIPTION
**Pull request by trel**
oops, missed this in my local diff.

added some comments.

same as https://github.com/irods/irods/pull/7307

### CODE REVIEW

File: scripts/setup_irods.py [hunk 1]
```diff
@@ -194,13 +194,20 @@ def check_hostname():
         raise IrodsError('The hostname (%s) must resolve to the local machine.' % (hostname))
 
 def determine_server_role(irods_config):
-    catalog_service_roles = set(['provider', 'consumer'])
+    # enum
```
Comment (korydraughn) [inline scripts/setup_irods.py:197]: I think this needs more words. How about ...
```py
# The list of catalog service roles supported by the server.
```

File: scripts/setup_irods.py [hunk 2]
```diff
@@ -194,13 +194,20 @@ def check_hostname():
...
+    catalog_service_roles = ['provider', 'consumer']
+
+    # get earlier setting, set to 'provider' if not set
```
Comment (korydraughn) [inline scripts/setup_irods.py:200]: How about ...
```py
# Get the service role previously configured if available. Otherwise, set to "provider".
```

**Review by korydraughn — COMMENTED** [review]
Just a few comments :-)

Overall, the code is fine. Arrays work!

File: scripts/setup_irods.py (hunk 1 above)
Comment (trel) [reply scripts/setup_irods.py:197]: certainly

File: scripts/setup_irods.py (hunk 2 above)
Comment (trel) [reply scripts/setup_irods.py:200]: will do

**Comment by trel** [comment]
updated

**Comment by trel** [comment]
squashed and #'d

**Review by alanking — APPROVED** [review]

**Pull request merged**